python/ray/services.py

```diff
@@ -559,12 +556,10 @@ def wait_for_redis_to_start(redis_ip_address,
             redis_client.client_list()
         except redis.ConnectionError:
             # Wait a little bit.
-            time.sleep(1)
-            logger.info("Failed to connect to the redis server, retrying.")
-            counter += 1
+            time.sleep(0.001)
         else:
             break
-    if counter == num_retries:
+    else:
```

Comment:
I was a little hesitant to use this syntax because it's not widely known and non-obvious what it does, but thought it was ok since the intended behavior should be clear from the error message.

Reply:
I like it. Very pythonic.